public/apps/configuration/utils/audit-logging-view-utils.tsx

```diff
@@ -25,60 +26,38 @@ import {
 } from '@elastic/eui';
 
 import React from 'react';
+import { HttpStart } from 'kibana/public';
 import { displayArray, displayBoolean, displayObject, renderTextFlexItem } from './display-utils';
-
-interface GeneralSettings {
-  // layer
-  enable_rest: boolean;
-  enable_transport: boolean;
-  disabled_rest_categories: string[];
-  disabled_transport_categories: string[];
-
-  // attribute
-  resolve_bulk_requests: boolean;
-  resolve_indices: boolean;
-  log_request_body: boolean;
-  exclude_sensitive_headers: boolean;
-
-  // ignore
-  ignore_users: string[];
-  ignore_requests: string[];
-}
-
-interface ComplianceSettings {
-  enabled: boolean;
-
-  // read
-  read_metadata_only: boolean;
-  read_ignore_users: string[];
-  read_watched_fields: {
-    [key: string]: string;
-  };
-
-  // write
-  write_metadata_only: boolean;
-  write_log_diffs: boolean;
-  write_ignore_users: string[];
-  write_watched_indices: string[];
-}
+import { API_ENDPOINT_AUDITLOGGING_UPDATE } from '../constants';
+import {
+  AuditLoggingSettings,
+  ComplianceSettings,
+  GeneralSettings,
+} from '../panels/audit-logging/types';
 
 export function renderStatusPanel(onSwitchChange: () => void, auditLoggingEnabled: boolean) {
```

Comment:
Is this render function re-usable? If not, I would suggest moving to the `public/apps/configuration/panels/audit-logging/` subfolder

Reply:
renderStatusPanel is shared for enabled and disabled situations. The original motivation is to keep audit-logging.tsx simple and put all subfunctions to support it in its util class.

Reply:
For non-sharing rendering function I think it would be better to put in the same folder with the component rather than in the public shared util folder.